src/rules/no-invalid-named-grid-areas.js

```diff
@@ -0,0 +1,174 @@
+ */
+const nullCellToken = /^\.+$/u;
+
+/**
+ * Finds non-rectangular grid areas in a 2D grid
+ * @param {string[][]} grid 2D array representing the grid areas
+ * @returns {Array<{name: string, row: number}>} Array of errors found
+ */
+function findNonRectangularAreas(grid) {
+	const errors = [];
+	const names = [...new Set(grid.flat())].filter(
+		name => !nullCellToken.test(name),
+	);
+
+	for (const name of names) {
+		const indicesByRow = grid.map(row => {
+			const indices = [];
+			let idx = row.indexOf(name);
+
+			while (idx !== -1) {
+				indices.push(idx);
+				idx = row.indexOf(name, idx + 1);
+			}
+
+			return indices;
+		});
+
+		for (let i = 0; i < indicesByRow.length; i++) {
+			for (let j = i + 1; j < indicesByRow.length; j++) {
+				const row1 = indicesByRow[i];
+				const row2 = indicesByRow[j];
+
+				if (row1.length === 0 || row2.length === 0) {
+					continue;
+				}
+
+				if (
+					row1.length !== row2.length ||
+					!row1.every((val, idx) => val === row2[idx])
+				) {
+					const firstRow = grid.findIndex(r => r.includes(name));
+					errors.push({ name, row: firstRow });
+				}
+			}
+		}
+	}
+
+	return errors;
+}
+
+const validProps = new Set(["grid-template-areas", "grid-template", "grid"]);
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+/** @type {NoInvalidNamedGridAreasRuleDefinition} */
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			description: "Disallow invalid named grid areas",
+			recommended: true,
+			url: "https://github.com/eslint/css/blob/main/docs/rules/no-invalid-named-grid-areas.md",
+		},
+
+		messages: {
+			emptyGridArea: "Grid area must contain at least one cell token.",
+			unevenGridArea:
+				"Grid area strings must have the same number of cell tokens.",
+			nonRectangularGridArea:
+				"Cell tokens with name '{{name}}' must form a rectangle.",
+		},
+	},
+
+	create(context) {
+		return {
+			Declaration(node) {
+				const propName = node.property.toLowerCase();
+
+				if (
+					validProps.has(propName) &&
+					node.value.type === "Value" &&
+					node.value.children.length > 0
+				) {
+					const stringNodes = node.value.children.filter(
+						child => child.type === "String",
+					);
+
+					if (stringNodes.length === 0) {
+						return;
+					}
+
+					const grid = [];
+					const emptyNodes = [];
+					const unevenNodes = [];
+					let firstRowLen = null;
+
+					for (const stringNode of stringNodes) {
+						const trimmedValue = stringNode.value.trim();
+
+						if (trimmedValue === "") {
+							emptyNodes.push(stringNode);
+							continue;
+						}
+
+						const row = trimmedValue.split(" ").filter(Boolean);
+						grid.push(row);
+
+						if (firstRowLen === null) {
+							firstRowLen = row.length;
+						} else if (row.length !== firstRowLen) {
+							unevenNodes.push(stringNode);
+						}
```

Comment:
Right now we don't report the `"a b"` line, but I think we should:

```css
/* eslint css/no-invalid-named-grid-areas: "error" */

.grid {
	grid-template-areas:
  "a b" // Should we report this as well since this is also uneven and we don't know if the intention is to have 2,3, or 4 cells
  "a b c" //  Grid area strings must have the same number of cell tokens 
  "a b c d"; //  Grid area strings must have the same number of cell tokens 
}
```

Reply:
For most linting rules we only flag the first pattern that is unexpected. In this example, the second row is unexpected because of the first row setting the expectation that a row has two cells. So I think the current behavior is inline with how we generally make lint rules work.